common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/azureactivedirectory/AzureActiveDirectoryAuthorizationResult.java

```diff
@@ -22,7 +22,117 @@
 import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationResult;
+import com.microsoft.identity.common.internal.util.StringUtil;
+
+import java.io.Serializable;
+import java.util.HashMap;
+
+import static com.microsoft.identity.common.adal.internal.AuthenticationConstants.Browser.RESPONSE_ERROR_CODE;
+import static com.microsoft.identity.common.adal.internal.AuthenticationConstants.Browser.RESPONSE_ERROR_MESSAGE;
+import static com.microsoft.identity.common.adal.internal.AuthenticationConstants.Browser.RESPONSE_FINAL_URL;
 
 public class AzureActiveDirectoryAuthorizationResult extends AuthorizationResult {
+
+    private static final String TAG = AzureActiveDirectoryAuthorizationResult.class.getSimpleName();
+
+    public AzureActiveDirectoryAuthorizationResult(int resultCode, final Intent data) {
+        if (data == null) {
+            AuthorizationErrorResponse errorResponse = new AuthorizationErrorResponse
+                    (AuthorizationErrorResponse.AUTHORIZATION_FAILED, AuthorizationErrorResponse.NULL_INTENT);
+            setAuthorizationErrorResponse(AuthorizationStatus.FAIL, errorResponse);
+            return;
+        }
+        final Bundle extras = data.getExtras();
+        final int requestId = extras.getInt(AuthenticationConstants.Browser.REQUEST_ID);
+        switch (resultCode) {
+
+            case AuthenticationConstants.UIResponse.BROWSER_CODE_CANCEL:
+                Logger.verbose(TAG, null, "User cancel the request in webview: " + requestId);
+                AuthorizationErrorResponse errorResponse = new AuthorizationErrorResponse
+                        (AuthorizationErrorResponse.USER_CANCEL, AuthorizationErrorResponse.USER_CANCELLED_FLOW);
+                setAuthorizationErrorResponse(AuthorizationStatus.USER_CANCEL, errorResponse);
+                break;
+
+            case AuthenticationConstants.UIResponse.BROWSER_CODE_COMPLETE:
+                final String url = extras.getString(RESPONSE_FINAL_URL, "");
+                parseAuthorizationResponse(url);
+                break;
+
+            case AuthenticationConstants.UIResponse.BROWSER_CODE_ERROR:
+                // This is purely client side error, possible return could be chrome_not_installed or the request intent is
+                // not resolvable
+                final String error = extras.getString(RESPONSE_ERROR_CODE);
+                final String errorDescription = extras.getString(RESPONSE_ERROR_MESSAGE);
+                setAuthorizationErrorResponse(AuthorizationStatus.FAIL, new AuthorizationErrorResponse(error, errorDescription));
+                break;
+
+            case AuthenticationConstants.UIResponse.BROWSER_CODE_AUTHENTICATION_EXCEPTION:
+                //Serializable authException = extras.getSerializable(AuthenticationConstants.Browser.RESPONSE_AUTHENTICATION_EXCEPTION);
+                //TODO : AuthenticationException is not part of Common, what needs to be done here.
```

Comment:
`Common` has its own `ClientException` and `ServiceException` classes -- take a look at those and determine which (if either) suits your needs and is semantically the most appropriate. If neither is, let's follow up